nameserver/nameserver.go

```diff
@@ -208,12 +214,16 @@ func (n *Nameserver) OnGossipUnicast(sender router.PeerName, msg []byte) error {
 }
 
 func (n *Nameserver) receiveGossip(msg []byte) (router.GossipData, router.GossipData, error) {
-	var entries Entries
-	if err := gob.NewDecoder(bytes.NewReader(msg)).Decode(&entries); err != nil {
+	var gossip GossipData
+	if err := gob.NewDecoder(bytes.NewReader(msg)).Decode(&gossip); err != nil {
 		return nil, nil, err
 	}
 
-	if err := entries.check(); err != nil {
+	if gossip.Timestamp < now()-int64(gossipWindow/time.Second) {
+		return nil, nil, fmt.Errorf("Refusing old gossip message: %d", gossip.Timestamp)
```

Comment:
Think it would be worth telling the user the apparent difference in time; it's quite hard from a 64-bit int to see whether your clock is just a bit slow or your machine is set to 1970.